### PR TITLE
feat(frontend): sort adv search and collection list in alphabetical o…

### PR DESCRIPTION
**Pull request by edalex-yinzi**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [x] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change
Create util file sort.ts
Add sortArrayOfObjects func
Add correspongding test file
Sort adv search and collection list in alphabetical order

image

image

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: react-front-end/tsrc/search/components/CollectionSelector.tsx [hunk 1]
```diff
@@ -57,7 +58,9 @@ export const CollectionSelector = ({
 
   useEffect(() => {
     collectionListSummary([OEQ.Acl.ACL_SEARCH_COLLECTION])
-      .then((collections: Collection[]) => setCollections(collections))
+      .then((collections: Collection[]) =>
+        setCollections(sortArrayOfObjects(collections, "name"))
```
Comment (edalex-ian) [inline react-front-end/tsrc/search/components/CollectionSelector.tsx:62]: Heya @edalex-yinzi ,

No need to write your own function here. You should be able to `A.sort`, `contramap`, a function like `({name}) => name` and then `S.Ord`.

Comment (PenghaiZhang) [reply react-front-end/tsrc/search/components/CollectionSelector.tsx:62]: Yeah. You can find an example from `UserModule.ts`.

**Comment by edalex-yinzi** [comment]
The only difference is S.ORD is case sensitive.

image

File: react-front-end/tsrc/search/components/CollectionSelector.tsx [hunk 2]
```diff
@@ -57,7 +62,14 @@ export const CollectionSelector = ({
 
   useEffect(() => {
     collectionListSummary([OEQ.Acl.ACL_SEARCH_COLLECTION])
-      .then((collections: Collection[]) => setCollections(collections))
+      .then((collections: Collection[]) =>
+        setCollections(
+          pipe(
+            collections,
+            A.sort(ORD.contramap((as: Collection) => as.name)(S.Ord))
```
Comment (PenghaiZhang) [inline react-front-end/tsrc/search/components/CollectionSelector.tsx:69]: The name of `as` is a bit confusing.
What about `ORD.contramap<string, Collection>(({name}) => name)`

File: react-front-end/tsrc/search/components/CollectionSelector.tsx [hunk 3]
```diff
@@ -22,12 +22,17 @@ import { Autocomplete, AutocompleteGetTagProps } from "@material-ui/lab";
...
+import { pipe } from "fp-ts/function";
+import * as ORD from "fp-ts/Ord";
+import * as S from "fp-ts/string";
 import { TooltipChip } from "../../components/TooltipChip";
 import {
   Collection,
   collectionListSummary,
 } from "../../modules/CollectionsModule";
 import { languageStrings } from "../../util/langstrings";
+
```
Comment (PenghaiZhang) [inline react-front-end/tsrc/search/components/CollectionSelector.tsx:35]:
```suggestion
```

File: react-front-end/tsrc/search/SearchPage.tsx [hunk 4]
```diff
@@ -864,7 +866,14 @@ const SearchPage = ({ updateTemplate, advancedSearchId }: SearchPageProps) => {
       title: searchStrings.advancedSearchSelector.title,
       component: (
         <AdvancedSearchSelector
-          advancedSearches={advancedSearches}
+          advancedSearches={pipe(
```
Comment (PenghaiZhang) [inline react-front-end/tsrc/search/SearchPage.tsx:869]: I would put this sorting in `AdvancedSearchSelector` rather than here.

**Comment by edalex-ian** [comment]
Looks like good feedback there @edalex-yinzi from @PenghaiZhang . I'll wait until you've addressed that to re-review.

**Comment by edalex-yinzi** [comment]
Thanks, @PenghaiZhang  and @edalex-ian.

**Review by edalex-ian — APPROVED** [review]

**Pull request merged**